src/extension/providers/postMessage.ts

```diff
@@ -0,0 +1,51 @@
+export function postMessage<T>(
+  scope: string,
+  action: string,
+  args: T | undefined
+): Promise<T> {
+  return new Promise((resolve, reject) => {
+    const id = Math.random().toString().slice(4);
+
+    // post the request to the content script. from there it gets passed to the background script and back
+    // in page script can not directly connect to the background script
```

Comment:
It seems this might be false and doesn't really make sense to me. When you post a message to the window, anything listening to the window should receive it. Are you able to add a TODO: comment to re-investigate this?